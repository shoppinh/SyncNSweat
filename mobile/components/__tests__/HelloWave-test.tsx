import * as React from 'react';
import { render } from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import { HelloWave } from '../HelloWave';

// Mock react-native-reanimated
jest.mock('react-native-reanimated', () => {
  const Reanimated = require('react-native-reanimated/mock');
  // The mock for `call` immediately calls the callback which is incorrect
  // So we override it with a no-op
  Reanimated.default.call = () => {};
  return Reanimated;
});

// Mock the useThemeColor hook
jest.mock('@/hooks/useThemeColor', () => ({
  useThemeColor: jest.fn().mockReturnValue('#000000'),
}));

describe('<HelloWave />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<HelloWave />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains the wave emoji', () => {
    const { getByText } = render(<HelloWave />);
    expect(getByText('ðŸ‘‹')).toBeTruthy();
  });

  it('starts the animation on mount', () => {
    // Create a spy on useEffect
    const useEffectSpy = jest.spyOn(React, 'useEffect');
    
    render(<HelloWave />);
    
    // Verify useEffect was called
    expect(useEffectSpy).toHaveBeenCalled();
    
    // Clean up
    useEffectSpy.mockRestore();
  });
});
