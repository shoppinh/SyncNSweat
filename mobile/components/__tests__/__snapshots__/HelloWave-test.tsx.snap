// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HelloWave /> renders correctly 1`] = `
<View
  style={
    {
      "transform": [
        {
          "rotate": "0deg",
        },
      ],
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#000000",
        },
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        {
          "fontSize": 28,
          "lineHeight": 32,
          "marginTop": -6,
        },
      ]
    }
  >
    ðŸ‘‹
  </Text>
</View>
`;
